hack/update-vendor.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+###
+#Copyright 2019 The KubeEdge Authors.
+#
+#Licensed under the Apache License, Version 2.0 (the "License");
+#you may not use this file except in compliance with the License.
+#You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#Unless required by applicable law or agreed to in writing, software
+#distributed under the License is distributed on an "AS IS" BASIS,
+#WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#See the License for the specific language governing permissions and
+#limitations under the License.
+###
+
+# The root of the build/dist directory
+KUBEEDGE_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
+
+#KUBEEDGE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+
+# list_staging_repos outputs a sorted list of repos in staging/src/kubeedge
+# each entry will just be the $repo portion of staging/src/kubeedge/$repo/...
+# $KUBEEDGE_ROOT must be set.
+function kubeedge::util::list_staging_repos() {
+  (
+    cd "${KUBEEDGE_ROOT}/staging/src/github.com/kubeedge/" && \
+    find . -mindepth 1 -maxdepth 1 -type d | cut -c 3- | sort
+  )
+}
+
+
+function kubeedge::git::check_status() {
+	modified=$( git status --short 2>/dev/null | wc -l)
+	echo $modified
+}
+ 
+
+go mod tidy
+ret=$(kubeedge::git::check_status)
+if [ ${ret} -eq 0 ]; then
+	echo "SUCCESS: go.mod and go.sum are in tiny"
+else
+	echo  "FAILED: go.mod / go.dum needs an update"
+	exit 1
+fi
+
+
+go mod vendor
+
+# create a symlink in vendor directory pointing to the staging components.
+# This lets other packages and tools use the local staging components as if they were vendored.
+for repo in $(kubeedge::util::list_staging_repos); do
+  rm -fr "${KUBEEDGE_ROOT}/vendor/github.com/kubeedge/${repo}"
+  ln -s "../../../staging/src/github.com/kubeedge/${repo}/" "${KUBEEDGE_ROOT}/vendor/github.com/kubeedge/${repo}"
```

Comment:
`"../../../staging/src/github.com/kubeedge/${repo}/"`
You don't know if this path exists on your local system. So it is possible, that this script could not be executed successfully because this path did not exists and your are not able to build kubeedge anymore.

Reply:
`../..` ?

Reply:
@subpathdev   i think`"../../../staging/src/github.com/kubeedge/${repo}/"` is ok,  actually，this is a relative path, and we already know the path relationship in advance.  for beehive(soft link) in vendor (vendor/github.com/kubeedge/beehive), it`s source file is `../../../staging/src/github.com/kubeedge/beehive`. it is no problem.

Reply:
Got it; I think it is easier to understand if you are using this:
`ln -s ${KUBEEDGE_ROOT}/staging/src/github.com/kubeedge/${repo} ./${repo}`

Reply:
This only works because you have executed a change directory in the function `kubeedge::util::list_staging_repos`

Reply:
@subpathdev `ln -s ${KUBEEDGE_ROOT}/staging/src/github.com/kubeedge/${repo} ${KUBEEDGE_ROOT}/vendor/github.com/kubeedge/${repo}` is wrong, beause `${KUBEEDGE_ROOT}` path is different for  users, and we can't use absolute paths.

Reply:
`kubeedge::util::list_staging_repos` did not change directory, because it exec in ().

Reply:
I have problems with the ln command; sry;